dnf/base.py

```diff
@@ -274,17 +275,19 @@ def close(self):
         self._tempfile_persistor = dnf.persistor.TempfilePersistor(
             self.conf.cachedir)
 
-        if self.clean_tempfiles:
+        if self.clean_trans_tempfiles:
             # delete all packages from last unsuccessful transactions
-            self._tempfiles.update(
+            self._trans_tempfiles.update(
                 self._tempfile_persistor.get_saved_tempfiles())
             self._tempfile_persistor.empty()
         if not self.conf.keepcache:
-            if self.clean_tempfiles:
-                self.clean_used_packages()
+            if self._tempfiles:
```

Comment:
I would personally remove unneeded `if` statement but if you find it more clear, keep it